nautobot/dcim/signals.py

```diff
@@ -286,3 +287,27 @@ def prevent_adding_tagged_vlans_with_incorrect_mode_or_site(sender, instance, ac
         return
 
     validate_interface_tagged_vlans(instance, kwargs["model"], kwargs["pk_set"])
+
+
+#
+# ControllerDeviceGroup
+#
+
+
+@receiver(post_save, sender=ControllerDeviceGroup)
+def handle_controller_device_group_controller_change(instance, raw=False, **_):
+    """Update descendants when the top level `ControllerDeviceGroup.controller` changes."""
+    if instance.parent or instance._original_controller == instance.controller:
+        return
+
+    logger = logging.getLogger(__name__ + ".ControllerDeviceGroup")
+
+    if raw:
+        logger.debug("Skipping controller update for imported controller device group %s", instance)
+        return
+
+    with transaction.atomic():
+        for group in instance.descendants(include_self=False):
```

Comment:
Unable to call `instance.descendants(include_self=False).update(controller=instance.controller`, fails with TreeModel related SQL error.